lib/validators/specValidator.js

```diff
@@ -568,7 +568,21 @@ class SpecValidator {
     let parameters = operation.getParameters();
     let result = { request: null, validationResult: { errors: [], warnings: [] } }, foundIssues = false;
     let options = {};
-
+    if (operation.pathObject && operation.pathObject.api && operation.pathObject.api.host) {
+      let scheme = 'https';
+      let basePath = '';
+      let host = operation.pathObject.api.host;
+      if (host.endsWith('/'))
+        host = host.slice(0, host.length - 1);
+      if (operation.pathObject.api.schemes && !operation.pathObject.api.schemes.some((item) => { return item && item.toLowerCase() === 'https'; }))
+        scheme = operation.pathObject.api.schemes[0];
+      if (operation.pathObject.api.basePath)
+        basePath = operation.pathObject.api.basePath;
+      if (!basePath.startsWith('/'))
+        basePath = `/${basePath}`;
+      let baseUrl = `${scheme}://${host}${basePath}`;
```

Comment:
does it matter if this url ends with "/" or not?, just checking, I've seen cases in the specs where the "basePath" is defined as "/" only.

Reply:
so if the basePath does not start with "/" only then we add forward slash.